doc/Programming-guide/Using-Kaa-endpoint-SDKs/C/SDK-Linux/index.md

```diff
@@ -2,304 +2,185 @@
 
-To build the C endpoint SDK, do the following:
+```
+sudo apt-get install cmake libssl-dev gcc
+```
 
-<!-- TODO: Fix this link when Administration-UI-guide#AdministrationUIguide-GeneratingSDK are updated -->
-1. [Generate ](Administration-UI-guide#AdministrationUIguide-GeneratingSDK)the C endpoint SDK in Admin UI.
-2. Download and untar the Kaa C SDK archive.
+### Build procedure
 
-   ~~~ bash
-   $ tar xfv kaa-c-ep-sdk.tar.gz
-   ~~~
+After you generated archive with C SDK, proceed as follows:
 
-3. Run the following commands.
+1. Unpack sdk:
 
-   ~~~ bash
-   $ mkdir build
-   $ cd build
-   $ cmake ..
-   $ make
-   $ make install
-   ~~~
+        tar -xvf c-sdk-archive-name.tar.gz
 
-## Configure C endpoint SDK
+2. Create the directory where sdk will be built:
 
-To configure the C endpoint SDK build, you can optionally specify the following parameters for the **cmake** command.
 
+        mkdir build
+        cd build
 
-{% include csv-table.html csv_path='cmake-c-options.csv' table_id='table-c-cmake-options' width='100%' %}
+3. Configure the build via cmake:
 
+        cmake ..
 
-**\* NOTE:**
+4. Perform build:
 
-Before running the **cmake** command with the KAA\_PLATFORM parameter for a platform other than supported, do the following:
+        make
 
-1. Create a folder in $KAA\_HOME/listfiles/platform/ and name it using \[a-zA-z\_-\] symbols. You will be able to use the folder name as a value for the KAA\_PLATFORM parameter.
-2. Put the CMakeLists.txt file into the created folder. This file may contain specific compilation/linking flags, platform-dependent source files, third-party library dependencies, that is all information necessary for building the C endpoint SDK for this platform.
-3. Optionally, specify the following parameters in the CMakeLists.txt file.
-  * KAA\_INCLUDE\_PATHS - full path(s) to folder(s) containing additional header files
-  * KAA\_SOURCE\_FILES - full path(s) to additional source files
-  * KAA\_THIRDPARTY\_LIBRARIES - third-party libraries (the name of the library, for example, ssl, crypto)
+## Building Kaa application
 
-The following example illustrates the build procedure for the debug build with the INFO log level and disabled EVENTS feature.
+This section is about how to build your Kaa aplication using C SDK.
 
-~~~
-mkdir build
-cd build
-cmake -DKAA_DEBUG_ENABLED=1 -DKAA_MAX_LOG_LEVEL=4 -DKAA_WITHOUT_EVENTS=1 ..
-make
-make install
-~~~
+In this guide we use *CMake* as a build system for the application, so focus on the writing the build system rather than the application's code. If you want master writing applociation code go to the [Programming guide]({{root_url}}/Programming-guide) section. Before continuing, make sure that all [dependencies](#dependencies) are installed.
 
-# C++ endpoint SDK
+1. First of all let's create some directories. The directories determine build structure. To keep it clean and transparent create three directories:
```

Comment:
The directories of the project determine the build structure. To keep it clean and transparent, let's create following directories: